llvm/tools/llvm-xargs/llvm-xargs.cpp

```diff
@@ -0,0 +1,140 @@
+//===- llvm-xargs.cpp - Command lines execution ---------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This source is utility to execute command lines. The specified command will
+// be invoked as many times as necessary to use up the list of input items.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/LineIterator.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Program.h"
+#include "llvm/Support/SystemUtils.h"
+
+using namespace llvm;
+
+static cl::opt<std::string> InputFileList{
+    cl::Positional, cl::Required, cl::desc("<input list of files>"),
+    cl::init("-"), cl::value_desc("filename")};
+
+static cl::opt<std::string> InputCommand{
+    cl::Positional, cl::Required, cl::desc("<command>"), cl::init("echo"),
+    cl::value_desc("command")};
+
+static cl::opt<std::string> Replace{
+    "in-replace",
+    cl::desc("Replace R in command with names read from standard input; if R "
+             "is unspecified, assume {}"),
+    cl::init("{}"), cl::value_desc("R")};
+
+static cl::opt<std::string> OutReplace{
+    "out-replace",
+    cl::desc("Replace R in command with temporary filename for writing outputs "
```

Comment:
Just by looking at the help message, I cannot figure out what this option does.

Reply:
Tried to improve.